components/sidebar_right/index.js

```diff
@@ -47,11 +47,12 @@ function mapStateToProps(state) {
         channel,
         currentUserId: getCurrentUserId(state),
         postRightVisible: Boolean(getSelectedPostId(state)),
-        searchVisible: Boolean(rhsState),
+        searchVisible: Boolean(rhsState) && rhsState !== RHSStates.PLUGIN,
         previousRhsState: getPreviousRhsState(state),
         isMentionSearch: rhsState === RHSStates.MENTION,
         isFlaggedPosts: rhsState === RHSStates.FLAG,
         isPinnedPosts: rhsState === RHSStates.PIN,
+        isPluginView: rhsState === RHSStates.PLUGIN,
```

Comment:
I wonder if we should just pass in `rhsState` as a prop instead of adding a series of boolean props based off of it. Just an observation and not something that needs to be changed though.

Reply:
It would probably clean up all those conditional calculated props and most are actually just passed directly to the SearchResults component. It may even make sense to map them in the state there directly given that they are use for that component's logic of what type of search to show. And just keep the logic to show search or plugin on these one.

Reply:
Let me know if you want me to move forward with this.

Reply:
You don't need to change it. It might be nice in the future, but my comment was just an observation.